WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift

```diff
@@ -431,6 +440,47 @@ private extension StoreStatsAndTopPerformersViewController {
         stackView.addArrangedSubview(separator)
         return stackView
     }
+
+    func startCustomRangeTabCreation() {
+        guard let navigationController else { return }
+        customRangeCoordinator = CustomRangeTabCreationCoordinator(
+            navigationController: navigationController,
+            onDateRangeSelected: { [weak self] start, end in
+                self?.createCustomRangeTab(start, end)
+            }
+        )
+        customRangeCoordinator?.start()
+    }
+
+    func createCustomRangeTab(_ start: Date, _ end: Date) {
+        let currentDate = Date()
+
+        // TODO: 11935 Add the correct data fetching and displaying based on custom range.
+        let customRangeVC = StoreStatsAndTopPerformersPeriodViewController(siteID: siteID,
+                                                                           timeRange: .custom(from: start, to: end),
+                                                                           currentDate: currentDate,
+                                                                           canDisplayInAppFeedbackCard: true,
+                                                                           usageTracksEventEmitter: usageTracksEventEmitter)
+
+        periodVCs.append(customRangeVC)
+        timeRanges.append(.custom(from: start, to: end))
+
+        let customRangeTabbedItem = TabbedItem(title: Localization.customRangeTabTitle,
+                                               viewController: customRangeVC,
+                                               accessibilityIdentifier: Constants.customRangeTabAcessibilityIdentifier)
```

Comment:
❓ Can we use the existing `"period-data-" + timeRange.rawValue + "-tab"` pattern to create `accessibilityIdentifier`? 

I am thinking that this will be useful when we save the custom range configuration to disk and recreate the custom range tab in `init`. What do you think?  

One more nice to have: It would be great if we could move initializing `TabbedItem` into a helper method and reuse that from `init` and `createCustomRangeTab`.

Reply:
identifier creation updated in https://github.com/woocommerce/woocommerce-ios/pull/11977/commits/2a26d87ffaff3240b2040af32a29ed3b545f76fc

> One more nice to have: It would be great if we could move initializing TabbedItem into a helper method and reuse that from init and createCustomRangeTab.

At first I tried this, but ran into a problem: The `TabbedItem` creation in `init` is done inside a closure. If I create a helper method inside `StoreStatsAndTopPerformersViewController` to create `TabbedItem` and call it, it creates this issue:

> 'self' captured by a closure before all members were initialized

The problem is that `periodVCs` is not initialized at that point, so the helper function can't be called yet.